### ISSUE TITLE
how can i waitFor() a return value?

### ISSUE THREAD

**Issue by modosc** [wix/redux-saga-tester#13]
i've got a saga like:
```
function* foo() {
  yield put({ type: FOO })
  return 'foo'
}
```
in regular code i can get the return value by doing `await store.runSaga(foo))`.  how can i do it with a `sagaTester`?

would it be possible to make the default `waitFor()` behavior such that it does this if you don't pass in an `action`?

**Comment by modosc** [comment]
also, fwiw i can get the return value by using this pattern:
```
const generator = foo()
foo.next().value // put({ type: FOO })
foo.next().value // 'foo'
```
but when i look at `calledActions()` i don't see the return value.

**Comment by gunters63** [comment]
Yeah, thats a good idea.

We have quite complex sagas in our current project. Returning the promise from .start() enables us to write integration tests with an initial redux state which run the saga to completion and then check a diff of the redux state in our tests.

```
it('should ...', () => {
    // setup initial redux state
    // Setup and start the saga
    return saga.done
        .then(() => {
            // Check final redux state
        })
    });
```

I have added a local copy of sagaTester to our project for now with your change included.
Hopefully that PR gets merged soon.

**Comment by modosc** [comment]
fixed by #17
